File: automator/src/main/java/com/stardust/automator/filter/StringContainsFilter.java
package com.stardust.automator.filter;

import com.stardust.automator.UiObject;

/**
 * Created by Stardust on 2017/3/9.
 */

public class StringContainsFilter extends DfsFilter {
    private final KeyGetter mKeyGetter;
    private final String mContains;

    StringContainsFilter(String contains, KeyGetter keyGetter) {
        mContains = contains;
        mKeyGetter = keyGetter;
    }

    @Override
    protected boolean isIncluded(UiObject nodeInfo) {
        String key = mKeyGetter.getKey(nodeInfo);
        return key != null && key.contains(mContains);
    }

    @Override
    public String toString() {
        return mKeyGetter.toString() + "Contains(\"" + mContains + "\")";
    }
}

File: automator/src/main/java/com/stardust/automator/filter/StringContainsFilter.kt
package com.stardust.automator.filter

import com.stardust.automator.UiObject

/**
 * Created by Stardust on 2017/3/9.
 */

class StringContainsFilter internal constructor(private val mContains: String, private val mKeyGetter: KeyGetter) : Filter {

    override fun filter(node: UiObject): Boolean {
        val key = mKeyGetter.getKey(node)
        return key != null && key.contains(mContains)
    }

    override fun toString(): String {
        return mKeyGetter.toString() + "Contains(\"" + mContains + "\")"
    }
}
